sql/core/src/main/scala/org/apache/spark/sql/execution/basicPhysicalOperators.scala

```diff
@@ -77,9 +77,40 @@ case class ProjectExec(projectList: Seq[NamedExpression], child: SparkPlan)
     }
   }
 
-  override def outputOrdering: Seq[SortOrder] = child.outputOrdering
+  override def outputOrdering: Seq[SortOrder] =
+    ProjectHelper.outputOrdering(projectList, child.outputOrdering, child)
 }
 
+object ProjectHelper {
+  /**
+   * Determins the outputOrdering property for [[ProjectExec]] and [[TakeOrderedAndProjectExec]]
```

Comment:
yea, it doesn't make sense, but does it cause any problems? I checked the code in `EnsureRequirements`, looks like it's ok to have some useless sort orders.